### PR TITLE
[GRPH-49] get_account_history fix

### PR DESCRIPTION
**Pull request by gladcow**
Backports Bitshares fix https://github.com/bitshares/bitshares-core/pull/398 and adds unit tests for this fix.

### CODE REVIEW

File: tests/tests/account_history_test.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+/*
+ * Copyright (c) 2019 PBSA, and contributors.
```
Comment (oxarbitrage) [inline tests/tests/account_history_test.cpp]: I think this should be placed on top in the line of Cryptonomex.

File: tests/tests/account_history_test.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+#include <graphene/app/database_api.hpp>
+#include <graphene/app/api.hpp>
+#include <graphene/chain/account_object.hpp>
+
+#include "../common/database_fixture.hpp"
+
+#include <fc/smart_ref_impl.hpp>
+#include <fc/crypto/digest.hpp>
+
+
```
Comment (oxarbitrage) [inline tests/tests/account_history_test.cpp]: I think we can avoid double newlines.

File: tests/tests/account_history_test.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+BOOST_AUTO_TEST_CASE(get_account_history) {
+   try {
+      graphene::app::history_api hist_api(app);
+
+      //account_id_type() do 3 ops
+      create_bitasset("USD", account_id_type());
+      auto dan_acc = create_account("dan");
+      auto bob_acc = create_account("bob");
+
+
```
Comment (oxarbitrage) [inline tests/tests/account_history_test.cpp]: I know this new lines are also in bitshares, from the code is coming from. I think we can remove them, also sent pull request at bitshares for it https://github.com/bitshares/bitshares-core/pull/1955

**Comment by gladcow** [comment]
I've addressed the issues found by @oxarbitrage ,  re-review,  please.

**Comment by gladcow** [comment]
I've redirected PR to `develop` branch. Re-review, please

File: tests/tests/history_api_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (c) 2019 PBSA, and contributors.
+ */
+/*
+ * Copyright (c) 2015 Cryptonomex, Inc., and contributors.
```
[the remark this answers is not in the document]
Comment (gladcow) [reply tests/tests/history_api_tests.cpp]: Fixed

[the remark this answers is not in the document]
Comment (gladcow) [reply tests/tests/history_api_tests.cpp]: Fixed

**Pull request merged**